plugins/cpu/fu-cpu-helper-cet-common.S

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2024 Richard Hughes <richard@hughsie.com>
+ * Copyright 2024 H.J. Lu <hjl.tools@gmail.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+
+	.text
+	.globl	fu_cpu_helper_cet_testfn1
+	.type	fu_cpu_helper_cet_testfn1, %function
+fu_cpu_helper_cet_testfn1:
+	.cfi_startproc
+#if defined(__x86_64__)
+	movq	(%rsp), %rax
+	addq	$8, %rsp
+	jmp	*%rax
+#elif defined(__i386__)
+	movl	(%esp), %eax
+	addl	$4, %esp
+	jmp	*%eax
+#else
+# error Unsupported CPU with shadow stack!
+#endif
```

Comment:
Because of this I think we should change the criteria for `build_fwupdcethelper` to only compile on x86 or x86_64.

Reply:
Unless that's implied (IE `-fcf-protection` doesn't exist on non-x86)

Reply:
CET is only enabled for 64-bit mode.

Reply:
> So we really should only build the helper for x86-64 targets then.

Yes.

Reply:
Can you add another clause to decide `build_fwupdcethelper` in your PR to check `host_cpu == 'x86_64'`

Reply:
> Can you add another clause to decide `build_fwupdcethelper` in your PR to check `host_cpu == 'x86_64'`

Can you send me the diff to do it?

Reply:
Don't worry about it, I did it in another PR (https://github.com/fwupd/fwupd/pull/7224).  Thanks!